test/test_cuda.py

```diff
@@ -583,9 +583,10 @@ def test_serialization_array_with_storage(self):
         self.assertTrue(isinstance(q_copy[0], torch.cuda.FloatTensor))
         self.assertTrue(isinstance(q_copy[1], torch.cuda.IntTensor))
         self.assertTrue(isinstance(q_copy[2], torch.cuda.FloatTensor))
-        self.assertTrue(isinstance(q_copy[3], torch.cuda.IntStorage))
+        self.assertTrue(isinstance(q_copy[3], torch.storage.TypedStorage))
```

Comment:
I don't remember if we were going to do `__instancecheck__` shenanigans to get the old spelling to work lol

Reply:
I don't think we talked about it before, but we can do that if you'd like

Reply:
This will save you having to return torch.FloatStorage directly in the bindings; you can unconditionally return a TypedStorage but instance check will still work. I think it will simplify this PR a bit. I'd consider doing it in this PR if you get stuck on bugs related to this machinery; otherwise it'd probably one of the first follow ups I'd want to do afterwards.